uproot/_connect/to_pandas.py

```diff
@@ -104,95 +105,82 @@ def futures2df(futures, outputtype, entrystart, entrystop, flatten, flatname):
-                i = 0
-                numentries = entrystop - entrystart
-                while i < numentries:
-                    entries[starts[i]:stops[i]] = i + entrystart
-                    subentries[starts[i]:stops[i]] = numpy.arange(stops[i] - starts[i])
-                    i += 1
-
-                df = outputtype(index=pandas.MultiIndex.from_arrays([entries, subentries], names=["entry", "subentry"]))
 
                 interpretation = interpretation.content
                 if isinstance(interpretation, asobj) and isinstance(interpretation.content, astable):
                     interpretation = interpretation.content
                 if isinstance(interpretation, astable) and isinstance(interpretation.content, asdtype):
                     interpretation = interpretation.content
 
-                if isinstance(interpretation, asdtype):
-                    if interpretation.todims == ():
-                        if interpretation.todtype.names is None:
-                            fn = flatname(name, None, ())
-                            df[fn] = array.flatten()
-                        else:
-                            for nn in interpretation.todtype.names:
-                                if not nn.startswith(" "):
-                                    fn = flatname(name, nn, ())
-                                    df[fn] = array[nn].flatten()
-                    else:
-                        for tup in itertools.product(*[range(x) for x in interpretation.todims]):
-                            if interpretation.todtype.names is None:
-                                fn = flatname(name, None, tup)
-                                df[fn] = array[(slice(None),) + tup].flatten()
-                            else:
-                                for nn in interpretation.todtype.names:
-                                    if not nn.startswith(" "):
-                                        fn = flatname(name, nn, tup)
-                                        df[fn] = array[nn][(slice(None),) + tup].flatten()
+                array = future()
 
-                else:
-                    fn = flatname(name, None, ())
-                    df[fn] = list(array.flatten())
+                if starts is None:
+                    starts = array.starts
+                    stops = array.stops
+                array = array.flatten()
+
+                length = len(array)
+
+            else:
+                array = future()
+
+            names.append(name)
+            interpretations.append(interpretation)
+            arrays.append(array)
+
+        if length is None:
+            length = len(arrays[0])
+
+        index = pandas.Index(numpy.arange(length))
+        df = outputtype(index=index)
+
+        for name, interpretation, array in zip(names, interpretations, arrays):
+
+            if isinstance(interpretation, asdtype):
+
+                if len(array) < length:
```

Comment:
I removed the `if len(array) < length` (the code that was inside the block always runs now), but please tell me what you were thinking in case I'm misinterpreting you.

Reply:
Eventually, I added a `needsbroadcast` boolean to specify exactly when to broadcast and when not to, based on the original `interpretation` of the branch. Sorry if I wrote the original `len(array) < length` and I'm blaming it on you—if I did, it was a mistake that happened to work in most cases.